### PR TITLE
Resource assigned services page implemented

### LINKED ISSUE #289: Resource - assigned services
Create new page under the resource manager - Assigned services. Should look and be the same as in old gui. Implement all functionality.

### PR DESCRIPTION
**Pull request by xpavlic**
Resolves #289
- Resource - Assigned services page implemented

- add, remove buttons with implemented dialogs and fuctionality added

### CODE REVIEW

File: apps/admin-gui/src/assets/i18n/en.json [hunk 1]
```diff
@@ -1146,6 +1154,21 @@
       "SUCCESS_MESSAGE": "Selected groups were assigned to resource",
       "CONFIGURE_GROUPS": "Configure group(s) before assign"
     },
+    "ASSIGN_SERVICE_TO_RESOURCE": {
+      "TITLE": "Assign service",
+      "FILTER_DESCRIPTION": "Find service",
+      "ADD_BUTTON": "Add",
+      "CANCEL_BUTTON": "Cancel",
+      "SUCCESS_MESSAGE": "Selected messages were assigned to resource"
```
Comment (xkureck) [inline apps/admin-gui/src/assets/i18n/en.json:1162]: *selected services

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<mat-spinner class="ml-auto mr-auto" *ngIf="loading"></mat-spinner>
+<div class="mt-3" *ngIf="!loading && assignedServices.length !== 0">
+  <app-services-list
+    (page)="pageChanged($event)"
+    [pageSize]="pageSize"
+    [services]="filteredServices"
+    [selection]="selected">
+  </app-services-list>
+</div>
+<app-alert *ngIf="assignedServices.length === 0 && !loading" color="warn">
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html:22]: this alert is already part of the service-list, so there is no need to have it here

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+</button>
+<button mat-flat-button color="warn" class="ml-2" [disabled]="selected.selected.length === 0" (click)="removeServices()">
+  {{'RESOURCE_DETAIL.ASSIGNED_SERVICES.REMOVE_SERVICE' | translate}}
+</button>
+<app-debounce-filter
+  [placeholder]="'RESOURCE_DETAIL.ASSIGNED_SERVICES.FILTER_SERVICES'"
+  (filter)="applyFilter($event)">
+</app-debounce-filter>
+<mat-spinner class="ml-auto mr-auto" *ngIf="loading"></mat-spinner>
+<div class="mt-3" *ngIf="!loading && assignedServices.length !== 0">
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html:14]: `mt-3`is around the service-list already so it doesnt have to be here

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <app-services-list
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html:15]: for some reason the styling of service-list is broken. The card around the table is missing. Try rebase. If that doesnt help change first row of service.list.component.html and change class to:
`class="card mt-3"`

File: apps/admin-gui/src/app/shared/components/dialogs/assign-service-to-resource-dialog/assign-service-to-resource-dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<div class="{{theme}}">
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/assign-service-to-resource-dialog/assign-service-to-resource-dialog.component.html:1]: image
this dialog is too big on my notebook. Please look on the other dialogs how to add dialog if the height of the dialog is too big for viewport.

File: apps/admin-gui/src/app/shared/components/dialogs/assign-service-to-resource-dialog/assign-service-to-resource-dialog.component.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  <app-debounce-filter
+    placeholder="{{'DIALOGS.ASSIGN_SERVICE_TO_RESOURCE.FILTER_DESCRIPTION' | translate}}"
+    (filter)="applyFilter($event)">
+  </app-debounce-filter>
+  <mat-spinner *ngIf="loading" class="mr-auto ml-auto"></mat-spinner>
+  <app-services-list
+    (page)="pageChanged($event)"
+    [pageSize]="pageSize"
+    *ngIf="!loading"
+    [services]="filteredServices"
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/assign-service-to-resource-dialog/assign-service-to-resource-dialog.component.html:12]: same here with filtering of the services
[service]="services"
[fileringValue]="filteringValue"

File: apps/admin-gui/src/app/shared/components/dialogs/remove-service-from-resource-dialog/remove-service-from-resource-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+
+  constructor(public dialogRef: MatDialogRef<RemoveServiceFromResourceDialogComponent>,
+              @Inject(MAT_DIALOG_DATA) public data: RemoveServiceFromResourceDialogData,
+              private notificator: NotificatorService,
+              private translate: TranslateService,
+              private resourcesManager: ResourcesManagerService) {
+  }
+
+  loading: boolean;
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/remove-service-from-resource-dialog/remove-service-from-resource-dialog.component.ts:28]: You probably wanted to use the loading when performing the removeServices action to hide the table and show the user that someting is doing, but forget to use it at all. Please add that ass well.

File: apps/admin-gui/src/app/shared/components/services-list/services-list.component.html [hunk 8]
```diff
@@ -85,13 +85,13 @@
...
   </div>
 </div>
 
 <mat-spinner *ngIf="exporting" class="ml-auto mr-auto"></mat-spinner>
 
-<app-alert *ngIf="dataSource.filteredData.length === 0 && (this.services !== undefined && this.services.length !== 0)" color="warn">
+<app-alert *ngIf="dataSource.filteredData.length === 0 && (this.services !== undefined && this.services.length !== 0)"
+           color="warn">
   {{'SHARED_LIB.UI.ALERTS.NO_FILTER_RESULTS_ALERT' | translate}}
 </app-alert>
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/services-list/services-list.component.html:97]: So now the message that there are no services is not showing properly(row 98-100). Please rewrite it with:
`<app-alert *ngIf="this.services === undefined || this.services.length === 0" color="warn">
  {{'SHARED.COMPONENTS.SERVICES_LIST.NO_SERVICES_WARNING' | translate}}
</app-alert>`

File: apps/admin-gui/src/app/shared/components/dialogs/remove-service-from-resource-dialog/remove-service-from-resource-dialog.component.html [hunk 9]
```diff
@@ -0,0 +1,38 @@
+<h1 mat-dialog-title>{{'DIALOGS.REMOVE_SERVICE_FROM_RESOURCE.TITLE' | translate}}</h1>
+<div mat-dialog-content>
+  <p>
+    {{'DIALOGS.REMOVE_SERVICE_FROM_RESOURCE.DESCRIPTION' | translate}}
+  </p>
+
+  <div class="font-weight-bold">
+    {{'DIALOGS.REMOVE_SERVICE_FROM_RESOURCE.ASK' | translate}}
+  </div>
+  <mat-spinner *ngIf="loading" class="mr-auto ml-auto"></mat-spinner>
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/remove-service-from-resource-dialog/remove-service-from-resource-dialog.component.html:10]: also the spinner needs theme so it have proper color. Look to other dialogs how they set theme to whole dialog

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+</perun-web-apps-immediate-filter>
+<mat-spinner class="ml-auto mr-auto" *ngIf="loading"></mat-spinner>
+  <app-services-list
+    (page)="pageChanged($event)"
+    [pageSize]="pageSize"
+    [services]="assignedServices"
+    [selection]="selected"
+    [filterValue]="filterValue"
+    *ngIf="!loading">
+  </app-services-list>
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html:21]: remove tab

**Comment by Vojtech-Sassmann** [comment]
> image
> This error appears when adding services to resources. Tbh I dunno if thats a bug or appropriate behaviour.  @Vojtech-Sassmann ?

Hi, this is an expected behaviour. When you try to assign a service, attribute checks are started and everything has to be correctly set.

**Comment by xpavlic** [comment]
> > image
> > This error appears when adding services to resources. Tbh I dunno if thats a bug or appropriate behaviour.  @Vojtech-Sassmann ?
> 
> Hi, this is an expected behaviour. When you try to assign a service, attribute checks are started and everything has to be correctly set.

File: apps/admin-gui/src/assets/i18n/en.json [hunk 11]
```diff
@@ -172,6 +172,13 @@
       "FILTER_GROUPS": "Filter groups",
       "NO_GROUPS": "No groups."
     },
+    "ASSIGNED_SERVICES": {
+      "TITLE": "Assigned services",
+      "ADD_SERVICE": "Add..",
```
Comment (Vojtech-Sassmann) [inline apps/admin-gui/src/assets/i18n/en.json:177]:
```suggestion
      "ADD_SERVICE": "Add",
```

**Pull request merged**